modules/dispatcher/dispatcher.c

```diff
@@ -1281,22 +1283,40 @@ mi_response_t *ds_mi_set(const mi_params_t *params,
 	return init_mi_result_ok();
 }
 
-mi_response_t *ds_mi_list(int full)
+mi_response_t *ds_mi_list(const mi_params_t *params, int full)
 {
+	str partition_name;
+	int filter_partition = 0;
 	mi_response_t *resp;
 	mi_item_t *resp_obj;
 	mi_item_t *parts_arr, *part_item;
-	ds_partition_t *part_it;
+	ds_partition_t *partition, *part_it;
 
 	resp = init_mi_result_object(&resp_obj);
 	if (!resp)
 		return 0;
 
+	if (try_get_mi_string_param(params, "partition", &partition_name.s, &partition_name.len) == 0)
+		filter_partition = 1;
+
+	if (filter_partition) {
+		if (partition_name.s==NULL)
+			return init_mi_error(500, MI_SSTR("partition param not found"));
+
+		partition = find_partition_by_name(&partition_name);
+
+		if (partition == NULL)
+			return init_mi_error(500, MI_SSTR(MI_UNK_PARTITION));
+	}
+
 	parts_arr = add_mi_array(resp_obj, MI_SSTR("PARTITIONS"));
 	if (!parts_arr)
 		goto error;
 
 	for (part_it = partitions; part_it; part_it = part_it->next) {
+		if (filter_partition && str_strcmp(&partition->name,&part_it->name))
```

Comment:
And here test if `partition` is NULL and if not, do the filtering. BTW, IMO, the partition matching can be directly done, by pointer, like `partition==part_it` as these are structures in shared mem.

Reply:
This should be addressed as well.